cvetools/cvesearch.go

```
@@ -148,12 +148,38 @@ func (cv *ScanTools) ScanImageData(data *share.ScanData) (*share.ScanResult, err
 		return result, err
 	}
 
+	var k8sAppString string
 	files := make(map[string]*detectors.FeatureFile)
 	for name, data := range pkgs {
 		files[name] = &detectors.FeatureFile{Data: data}
+		if name == scan.Kubernetes {
+			var kubeResource scan.KubernetesResource
+			if err := json.Unmarshal(data, &kubeResource); err != nil {
+				log.WithFields(log.Fields{"name": name, "data": string(data)}).Error()
+			}
+			k8sAppString = kubeResource.Name
+		}
 	}
-	appPkgs := scan.NewScanApps(true).DerivePkg(pkgs)
 
+	appPkgs := scan.NewScanApps(true).DerivePkg(pkgs)
+	if k8sAppString != "" {
+		// append k8s repo into the appPkgs
+		if i := strings.Index(k8sAppString, ":"); i > 0 {
+			// remove the RKE variants
+			name := strings.TrimPrefix(k8sAppString[:i], "rke2-")
+			name = strings.TrimPrefix(name, "rke-")
+			name = fmt.Sprintf("go:k8s.io/%s", name)
+			version := strings.TrimPrefix(k8sAppString[i+1:], "v")
+			k8sAppRes := scan.AppPackage{
+				AppName:    "golang",        // belong to golang type
+				ModuleName: name,            // image name
+				Version:    version,         // tag
+				FileName:   scan.Kubernetes, // high-level k8s scan
+			}
+			log.WithFields(log.Fields{"k8sAppRes": k8sAppRes}).Debug()
+			appPkgs = append(appPkgs, k8sAppRes)
+		}
```

We should not eat the error if the `k8sAppString` is not in the right format.

---

If it is reasonable, please provide an error.

---

```
errors.New("invalid k8sAppString")
```